cms/djangoapps/contentstore/helpers.py

```diff
@@ -471,3 +493,31 @@ def is_item_in_course_tree(item):
         ancestor = ancestor.get_parent()
 
     return ancestor is not None
+
+
+def _get_usage_key_from_node(node, parent_id: str) -> UsageKey | None:
+    """
+    Returns the UsageKey for the given node and parent ID.
+
+    If the parent_id is not a valid UsageKey, or there's no "url_name" attribute in the node, then will return None.
+    """
+    try:
+        parent_key = UsageKey.from_string(parent_id)
+        # We're constructing a CourseKey here because they support make_usage_key,
+        # but this will work even if the parent context_key is a library,
+        # cf opaque_keys test_lib_key_make_usage_key
+        parent_context = CourseKey.from_string(str(parent_key.context_key))
```

Comment:
Perhaps I'm missing something here... why is this not just `parent_context = parent_key.context_key` ?

Reply:
mypy complains if I do it that way, because only CourseKey has `make_usage_key` defined:

> error: "LearningContextKey" has no attribute "make_usage_key"  [attr-defined]

But since we're not type checking this file yet, and [`ImportIdGenerator` does the same thing](https://github.com/open-craft/edx-platform/blob/jill/no-tags-in-olx/cms/djangoapps/contentstore/helpers.py#L238) earlier in the file, I agree, simpler is better :)

6c05c6e3a1